src/com/sap/piper/tools/JavaArchiveDescriptor.groovy

```diff
@@ -0,0 +1,62 @@
+package com.sap.piper.tools
+
+import com.sap.piper.EnvironmentUtils
+import com.sap.piper.FileUtils
+
+import hudson.AbortException
+
+
+class JavaArchiveDescriptor extends ToolDescriptor {
+
+    final javaTool
+    final javaOptions
+
+    JavaArchiveDescriptor(name, environmentKey, stepConfigurationKey, executablePath, executableName, version, versionOption, javaTool, javaOptions) {
+        super(name, environmentKey, stepConfigurationKey, executablePath, executableName, version, versionOption)
+        this.javaTool = javaTool
+        this.javaOptions = javaOptions
+    }
+
+    @Override
+    def getHome(script, configuration, log = true) {
+
+        def home
+        if (EnvironmentUtils.isEnvironmentVariable(script, environmentKey)) {
+            home = EnvironmentUtils.getEnvironmentVariable(script, environmentKey)
+            if (log) script.echo "$name home '$home' retrieved from environment."
+        }
+        else if (configuration.containsKey(stepConfigurationKey)) {
+            home = configuration.get(stepConfigurationKey)
+            if (log) script.echo "$name home '$home' retrieved from configuration."
+        } else if (isOnCurrentWorkingDirectory(script)){
+            home = ''
+            if (log) script.echo "$name expected on current working directory."
+        } else {
+            throw new AbortException(getConfigurationOptions())
+        }
+        return home
+    }
+
+    @Override
+    def getExecutable(script, configuration, log = true) {
+
+        def tool = getTool(script, configuration, log)
+        def javaExecutable = javaTool.getExecutable(script, configuration, false)
+        def executable = "$javaExecutable $javaOptions $tool"
```

Comment:
If I get it right the code here works only if the last argument inside ```javaOptions``` is ```-jar```. IMO we should not expect ```-jar``` provided from the caller, but hard code it here. The use-case for that class here is simply: call ```java -jar```, maybe with additional options.